imageio/core/imopen.py

```diff
@@ -0,0 +1,254 @@
+from numbers import Number
+import re
+import numpy as np
+
+from .format import FormatManager, MODENAMES
+from .request import Request, RETURN_BYTES
+from .util import Singleton
+
+
+class _imopen(metaclass=Singleton):
+    def __init__(self):
+        self._known_plugins = list()
+
+    def __call__(self, uri, *args, plugin=None, api='legacy', **kwargs):
+        if api == "legacy":
+            return LegacyPlugin(uri, *args, plugin=plugin, **kwargs)
```

Comment:
Using a unique `return` statement at the end of the function may ease understanding the logic :wink: